Shared/Actions/CreateMenuItem.swift

```diff
@@ -0,0 +1,507 @@
+							\.$data
+						}
+					}
+				}
+
+				DefaultCase {
+					Summary("Create Item with \(\.$menuTitle) and \(\.$subtitle)") {
+						\.$iconType
+						\.$backgroundShape
+						\.$background
+						\.$data
+					}
+				}
+			}
+		} otherwise: {
+			// MARK: No Title
+			Summary("Create Menu Item with \(\.$menuTitle)")
+		}
+	}
+
+	@Parameter(title: "Title")
+	var menuTitle: String
+
+	@Parameter(title: "Subtitle")
+	var subtitle: String?
+
+	@Parameter(
+		title: "Icon",
+		default: .sfSymbol
+	)
+	var iconType: RMIconType
+
+	@Parameter(
+		title: "Background",
+		description:
+"""
+A Background for your Icon
+
+Use this in combination with background shape to show a background behind your icon
+""",
+		default: .default
+	)
+	var background: RMStyle
+
+	// SF Symbol
+	@Parameter(
+		title: "SF Symbol Name",
+		description: """
+  The name of a SF Symbol
+
+  For available symbols see Apple's website (https://developer.apple.com/sf-symbols/)
+  """,
+		default: "plus"
+	)
+	var systemName: String
+
+	@Parameter(
+		title: "Foreground",
+		description: "The color for your SF Symbol",
+		default: .default
+	)
+	var foreground: RMStyle
+
+	// Emoji
+	@Parameter(
+		title: "Emoji",
+		description:
+"""
+Any Emoji 😀.
+
+Tap the Emoji button on your keyboard and select one emoji.
+""",
+		inputOptions: .init(keyboardType: .default)
+	)
+	var emoji: String?
+
+	@Parameter(
+		title: "Data"
+	)
+	var data: String?
+
+	@Parameter(
+		title: "Background Style",
+		description:
+"""
+The style of the icon's background.
+""",
+		default: .circle
+	)
+	var backgroundShape: RMBackgroundShape
+
+	func makeIcon() async -> Data? {
+		switch iconType {
+		case .sfSymbol:
+			return await RMIconContainer(
+				sfSymbol: systemName,
+				foregroundColor: foreground.color(),
+				backgroundColor: background.color(isBackground: true),
+				backgroundShape: backgroundShape
+			)
+			.render()
+		case .emoji:
+			return await RMIconContainer(
+				emoji: emoji ?? "",
+				backgroundColor: background.color(isBackground: true),
+				backgroundShape: backgroundShape
+			)
+			.render()
+		}
+	}
+
+	func perform() async throws -> some IntentResult & ReturnsValue<MenuItem>  {
+		let icon = await makeIcon()
+
+		let item = MenuItem(
+			title: menuTitle,
+			subtitle: subtitle,
+			icon: icon
+		)
+
+		return .result(value: item)
+	}
+}
+
+// MARK: Menu Item
+struct MenuItem: TransientAppEntity {
+	init() {
+		self.init(title: nil, subtitle: nil)
+	}
+
+	static let typeDisplayRepresentation: TypeDisplayRepresentation = "Menu Item"
+
+	var displayRepresentation: DisplayRepresentation {
+		let title: LocalizedStringResource = "\(title ?? "")"
+		let subtitle: LocalizedStringResource? = if let subtitle  { "\(subtitle)" } else { nil }
+		let image: DisplayRepresentation.Image? = if let icon {
+			if #available(iOS 17.0, macOS 14.0, *) {
+				.init(data: icon.data, displayStyle: .default)
+			} else {
+				.init(data: icon.data)
+			}
+		} else {
+			nil
+		}
+
```

Comment:
If the user specifies using a SF Symbol without any background, I think we should use `DisplayRepresentation.Image(systemName:, tintColor:)`.

Reply:
i disagree, because this would mean

- different sized icons when a menu of mixed background/no background items is displayed
- generating icon data in the AppEntity -> this is an asynchronous process, or
- generating icon data even though it isnt used AND difference between displayed icon & the icon that can be accessed by the user from the entity

Reply:
> different sized icons when a menu of mixed background/no background items is displayed

Good point. I didn't think about mixing styles.